src/lib/components/buttons/SubmitButtonComponent.svelte

```diff
@@ -1,8 +1,15 @@
 <script lang="ts">
-	import { Button } from 'flowbite-svelte';
+	import { Button, Spinner } from 'flowbite-svelte';
 	export let placeholder: any;
 	export let styles: string = '';
 	export let onClick: ((event: MouseEvent) => void) | undefined = undefined;
```

Comment:
Acá la función onclick solo  debería trabajar si loading is false. Resolver esto requiere que en vez de exportar la funcion dispares un evento y lo gestiones en el padre. Así si está loading no disparas el evento. No hay que hacer estas cosa ahora, incluso puede ser que sea mejor como está y lo controlas en el padre. Son solo comentarios para reflexionar.